tests/util/test_dt.py

```diff
@@ -164,3 +164,128 @@ def test_get_age(self):
 
         diff = dt_util.now() - timedelta(minutes=365*60*24)
         self.assertEqual(dt_util.get_age(diff), "1 year")
+
+    def test_parse_time_expression(self):
+        """Test parse_time_expression."""
+        self.assertEqual(
+            [x for x in range(60)],
+            dt_util.parse_time_expression('*', 0, 59)
+        )
+        self.assertEqual(
+            [x for x in range(60)],
+            dt_util.parse_time_expression(None, 0, 59)
+        )
+
+        self.assertEqual(
+            [x for x in range(0, 60, 5)],
+            dt_util.parse_time_expression('/5', 0, 59)
+        )
+
+        self.assertEqual(
+            [1, 2, 3],
+            dt_util.parse_time_expression([2, 1, 3], 0, 59)
+        )
+
+        self.assertEqual(
+            [x for x in range(24)],
+            dt_util.parse_time_expression('*', 0, 23)
+        )
+
+        self.assertEqual(
+            [42],
+            dt_util.parse_time_expression(42, 0, 59)
+        )
+
+        self.assertRaises(ValueError, dt_util.parse_time_expression, 61, 0, 60)
+
+    def test_find_next_time_expression_time_basic(self):
```

Comment:
One day it's time you learn how to write the modern style tests 😉

Reply:
Testing? That's overrated 😉 Just write a bunch of changes, publish a new version and hope nothing breaks #esphomelib 😅

I really should learn that :)